fhir-model/src/test/java/com/ibm/fhir/model/visitor/test/EncodingVisitorTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.model.visitor.test;
+
+import static com.ibm.fhir.model.type.String.string;
+import static org.testng.Assert.assertFalse;
+
+import java.io.StringWriter;
+import java.util.Collections;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.generator.FHIRGenerator;
+import com.ibm.fhir.model.resource.OperationOutcome;
+import com.ibm.fhir.model.resource.OperationOutcome.Issue;
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.model.type.CodeableConcept;
+import com.ibm.fhir.model.type.Extension;
+import com.ibm.fhir.model.type.Narrative;
+import com.ibm.fhir.model.type.Xhtml;
+import com.ibm.fhir.model.type.code.NarrativeStatus;
+import com.ibm.fhir.model.type.code.ResourceType;
+import com.ibm.fhir.model.visitor.EncodingVisitor;
+import com.ibm.fhir.model.visitor.EncodingVisitor.EncodingContext;
+
+public class EncodingVisitorTest {
+    private static OperationOutcome oo;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        oo = TestUtil.getMinimalResource(ResourceType.OPERATION_OUTCOME);
+        Issue issue = oo.getIssue().get(0).toBuilder()
+                .extension(Extension.builder()
+                    .id("\" onload=\"alert('test1')")
+                    .url("http://example.com/evil")
+                    .value(string("<script>alert('powned')</script>"))
+                    .build())
+                .details(CodeableConcept.builder()
+                    .text(string("<body onload=alert('powned2')>"))
+                    .build())
+                .diagnostics(string("<META HTTP-EQUIV=\"refresh\"\n"
+                        + "CONTENT=\"0;url=data:text/html;base64,PHNjcmlwdD5hbGVydCgndGVzdDMnKTwvc2NyaXB0Pg\">"))
+                .build();
+
+        oo = oo.toBuilder()
+                .text(Narrative.builder()
+                    .id("\" onmouseover=\"alert('Wufff!')")
+                    .extension(Collections.singleton(Extension.builder()
+                        .url("http://example.com/evil")
+                        .value(string("<script>alert('powned')</script>"))
+                        .build()))
+                    .status(NarrativeStatus.EMPTY)
+                    .div(Xhtml.from("Narrative text is tested elsewhere"))
+                    .build())
+                .issue(Collections.singleton(issue))
+                .build();
+    }
+
+    @Test
+    public void testEscape() throws Exception {
+        EncodingVisitor<OperationOutcome> v = new EncodingVisitor<>(EncodingContext.HTML_CONTENT);
```

Comment:
Where is EncodingVisitor.java called in the src/main/java? I didn't find the clear spot. it looks like Encode.forHtml is called directly in some spots.

Reply:
Only one spot.  In exceptionResponse(OperationOutcome, Status) I replace the Encode.forHtml of the diagnostic field with the EncodingVisitor which will encode all non-narrative string content in the resource:  https://github.com/IBM/FHIR/pull/2594/files#diff-9045cda3af019e8ce706c94d0064fce71e63dcf090b7bbd0c3931fb6a62b3066R310

Reply:
> it looks like Encode.forHtml is called directly in some spots.
Yeah, I left Encode.forHtml in fhir-provider because I think that is outside of the normal FHIRResource.exceptionResponse.  We could potentially use it there as well, but instead I'm leaning toward going with the approach of encoding only the error message that contain user content.
I *think* the fhir-provider error message could contain such context, but I'm not positive.